### ISSUE TITLE
Instability with the sentiment analysis example

### ISSUE THREAD

**Issue by vwxyzjn** [lvwerra/trl#417]
Hi all, I ran the latest TRL (after merging #410) with the sentiment analysis example for ten random seeds. I noticed 3 out 10 experiments would experience some amount of negative KL explosion, though one of them recovered 🫠...  Related https://github.com/lvwerra/trl/issues/235#issuecomment-1581370580

```
pip install openrlbenchmark==0.2.1a2
python -m openrlbenchmark.rlops_trl_multi_metrics \
    --filters '?we=costa-huang&wpn=trl&ceik=tracker_project_name&cen=log_with&metrics=env/reward_mean&metrics=env/reward_std&metrics=objective/kl_coef&metrics=objective/kl&metrics=objective/entropy&metrics=ppo/std_scores&metrics=ppo/mean_scores&metrics=ppo/learning_rate&metrics=ppo/mean_non_score_reward&metrics=ppo/loss/value&metrics=ppo/loss/total&metrics=ppo/loss/policy&metrics=ppo/policy/advantages_mean&metrics=ppo/policy/approxkl&metrics=ppo/policy/clipfrac&metrics=ppo/policy/entropy&metrics=ppo/returns/mean&metrics=ppo/returns/var' \
        'wandb?tag=gpt2-sentiment&cl=sentiment analysis (PR-410)' \
    --env-ids trl \
    --check-empty-runs \
    --pc.ncols 5 \
    --pc.ncols-legend 1 \
    --output-filename static/0compare \
    --scan-history
```

<img width="1173" alt="image" src="https://github.com/lvwerra/trl/assets/5555347/0879fe09-1bdb-45c3-bcea-04e6b1d04180">

<img width="387" alt="image" src="https://github.com/lvwerra/trl/assets/5555347/c696024d-2c33-4dd6-9b1d-37694ceb1b56">

**Comment by lvwerra** [comment]
Can you try without batched generation (in case you are using it)?

**Comment by lvwerra** [comment]
This should have been solved with #487 so closing for now.
